mlflow/models/model.py

```diff
@@ -1148,10 +1148,11 @@ def get_model_info(model_uri: str) -> ModelInfo:
     from mlflow.pyfunc import _download_artifact_from_uri
     from mlflow.store.artifact.models_artifact_repo import ModelsArtifactRepository
 
+    resolved_uri = model_uri
     if ModelsArtifactRepository._is_logged_model_uri(model_uri):
-        model_uri = ModelsArtifactRepository.get_underlying_uri(model_uri)
+        resolved_uri = ModelsArtifactRepository.get_underlying_uri(model_uri)
 
-    meta_file_uri = model_uri.rstrip("/") + "/" + MLMODEL_FILE_NAME
+    meta_file_uri = resolved_uri.rstrip("/") + "/" + MLMODEL_FILE_NAME
```

Comment:
Change to preserve the original model URI so `get_model_info(model_uri).model_uri == model_uri` is True.